From 39a28df904fca53bcee07dfe502237e4e282626b Mon Sep 17 00:00:00 2001
From: "FeRD (Frank Dana)" <ferdnyc@gmail.com>
Date: Sun, 27 Dec 2020 09:38:48 -0500
Subject: [PATCH] Backport fix for Unity progress crashes

The Ubuntu Unity integration is causing crashes with Python 3.9,
and since it's useless under Fedora anyway, we can just remove it.
---
 src/windows/main_window.py | 36 ++----------------------------------
 1 file changed, 2 insertions(+), 34 deletions(-)

diff --git a/src/windows/main_window.py b/src/windows/main_window.py
index d353596..8b21ec0 100644
--- a/src/windows/main_window.py
+++ b/src/windows/main_window.py
@@ -2415,36 +2415,11 @@ class MainWindow(QMainWindow, updates.UpdateWatcher):
         self.cache_object = new_cache_object
 
     def FrameExported(self, title_message, start_frame, end_frame, current_frame):
-        """Connect to Unity launcher (for Linux)"""
-        try:
-            if sys.platform == "linux" and self.has_launcher:
-                if not self.unity_launchers:
-                    # Get launcher only once
-                    import gi
-                    gi.require_version('Unity', '7.0')
-                    from gi.repository import Unity
-                    self.unity_launchers.append(Unity.LauncherEntry.get_for_desktop_id("openshot-qt.desktop"))
-                    self.unity_launchers.append(Unity.LauncherEntry.get_for_desktop_id("appimagekit-openshot-qt.desktop"))
-
-                # Set progress and show progress bar
-                for launcher in self.unity_launchers:
-                    launcher.set_property("progress", current_frame / (end_frame - start_frame))
-                    launcher.set_property("progress_visible", True)
-
-        except Exception:
-            # Just ignore
-            self.has_launcher = False
+        pass
 
     def ExportFinished(self, path):
         """Export has completed"""
-        try:
-            if sys.platform == "linux" and self.has_launcher:
-                for launcher in self.unity_launchers:
-                    # Set progress on Unity launcher and hide progress bar
-                    launcher.set_property("progress", 0.0)
-                    launcher.set_property("progress_visible", False)
-        except Exception:
-            pass
+        pass
 
     def transformTriggered(self, clip_id):
         """Handle transform signal (to keep track of whether a transform is happening or not)"""
@@ -2677,13 +2652,6 @@ class MainWindow(QMainWindow, updates.UpdateWatcher):
         # Create tutorial manager
         self.tutorial_manager = TutorialManager(self)
 
-        # Connect to Unity DBus signal (if linux)
-        if sys.platform == "linux":
-            self.unity_launchers = []
-            self.has_launcher = True
-            self.ExportFrame.connect(self.FrameExported)
-            self.ExportEnded.connect(self.ExportFinished)
-
         # Save settings
         s.save()
 
-- 
2.29.2

